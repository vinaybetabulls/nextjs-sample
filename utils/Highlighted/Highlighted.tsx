import React from 'react'
import _escapeRegExp from 'lodash/escapeRegExp'

export const Highlighted = ({ text = '', highlight = '' }) => {
  if (!highlight.trim()) {
    return <span>{text}</span>
  }
  const regex = new RegExp(`(${_escapeRegExp(highlight)})`, 'gi')
  const parts = text.split(regex)
  return (
    <span>
      {parts
        .filter((part) => part)
        .map((part, i) => (regex.test(part) ? <mark key={i}>{part}</mark> : <span key={i}>{part}</span>))}
    </span>
  )
}

export default Highlighted
