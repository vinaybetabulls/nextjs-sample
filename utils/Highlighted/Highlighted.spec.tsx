import React from 'react'
import { render } from '@testing-library/react'
import Highlighted from './Highlighted'

describe('Highlighted snapshot tests', () => {
  it('renders the correct snapshot for Highlighted text', () => {
    const { container } = render(<Highlighted text="testing" highlight="test" />)
    expect(container).toMatchSnapshot()
  })
  it('renders the correct snapshot when the highlighted text does not match', () => {
    const { container } = render(<Highlighted text="testing" highlight="sorry" />)
    expect(container).toMatchSnapshot()
  })
})
