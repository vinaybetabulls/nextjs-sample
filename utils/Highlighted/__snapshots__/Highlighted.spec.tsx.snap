// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Highlighted snapshot tests renders the correct snapshot for Highlighted text 1`] = `
<div>
  <span>
    <mark>
      test
    </mark>
    <span>
      ing
    </span>
  </span>
</div>
`;

exports[`Highlighted snapshot tests renders the correct snapshot when the highlighted text does not match 1`] = `
<div>
  <span>
    <span>
      testing
    </span>
  </span>
</div>
`;
